src/vs/workbench/common/editor.ts

```diff
@@ -168,6 +168,12 @@ export interface IFileEditorInputFactory {
 	isFileEditorInput(obj: unknown): obj is IFileEditorInput;
 }
 
+interface ICustomEditorInputFactory {
+	createCustomEditorInput(resource: URI, instantiationService: IInstantiationService): IEditorInput;
+
+	isBackingResourceForCustomEditor(resource: URI): boolean;
```

Comment:
@mjbvz I find this a bit too complicated, imho we can just agree to introduce a scheme called `custom` and treat it like that in the same way as we treat `untitled` and `file` in a special way.